### PR TITLE
[Merged by Bors] - feat(LinearAlgebra/ExteriorPower): the universal property of the exterior power

### PR DESCRIPTION
**Pull request by joelriou**
We obtain the universal property of the `n`th exterior power of a module `M`: linear maps from this exterior power identify to `n`-alternating maps from `M`. This is also phrased as the data of a presentation of the exterior power by generators and relations.

Co-authored-by: sophie.morel@ens-lyon.fr

---

This is partly extracted from #10654, with some modifications. Mostly, what was named `liftAlternatingEquiv` is `alternatingMapLinearEquiv` here. 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/ExteriorPower/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    [AddCommGroup M] [Module R M] :
+    Module.Relations R where
+  G := ι → M
+  R := Rels R ι M
+  relation r := match r with
+    | .add m i x y => Finsupp.single ((update m i x)) 1 +
+        Finsupp.single ((update m i y)) 1 -
+        Finsupp.single ((update m i (x + y))) 1
+    | .smul m i r x => Finsupp.single ((update m i (r • x))) 1 -
+        r • Finsupp.single ((update m i x)) 1
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/ExteriorPower/Basic.lean:93]: style nit: all these parens are unnecessarily doubled.

**Review by jcommelin — COMMENTED** [review]
Apart from the stylistic issue pointed out by Eric, I think this all looks very reasonable. But I would be happy to hear more thoughts from Eric.

File: Mathlib/LinearAlgebra/ExteriorPower/Basic.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/LinearAlgebra/ExteriorPower/Basic.lean]: Thanks for this observation!

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/LinearAlgebra/Span/Basic.lean [hunk 2]
```diff
@@ -80,6 +80,57 @@ theorem span_preimage_le (f : F) (s : Set M₂) :
 
 alias _root_.LinearMap.span_preimage_le := Submodule.span_preimage_le
 
+section
+
+variable {N : Type*} [AddCommMonoid N] [Module R N]
+
+lemma linearMap_eq_iff_of_eq_span {V : Submodule R M} (f g : V →ₗ[R] N)
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/Span/Basic.lean:87]: Should this be called `ext_iff_of_eq_span` or something?

Comment (joelriou) [reply Mathlib/LinearAlgebra/Span/Basic.lean:87]: I do not think so.

**Comment by joelriou** [comment]
Thanks!

bors merge

File: Mathlib/LinearAlgebra/Span/Basic.lean (hunk 2 above)
Comment (alreadydone) [reply Mathlib/LinearAlgebra/Span/Basic.lean:87]: > Should this be called `LinearMap.submodule_ext_iff_of_eq_span` or something?

I like this name too. It enables dot notation, and the iff version of extensionality lemmas are usually called `ext_iff`, e.g. [MvPolynomial.funext_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/MvPolynomial/Funext.html#MvPolynomial.funext_iff), [DFunLike.ext_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/FunLike/Basic.html#DFunLike.ext_iff)

**Pull request closed without merging**